### PR TITLE
Make getKoFileDeclarations more permissive

### PR DESCRIPTION
**Pull request by chrisbanes**
We've seen a lot of test flakiness lately with Konsist, due to it trying to parse generated files during builds. Since we can't control when Konsist tests are ran during a parallel Gradle build, Konsist can very easily get into a state where it tries to read generated Kotlin files whilst they're being generated.

This PR makes getKoFileDeclarations() more permissive, so that it doesn't throw "File must be a Kotlin file" exceptions for files which disappear in between the initial filetree walk and parsing.

### CODE REVIEW

File: lib/src/main/kotlin/com/lemonappdev/konsist/core/provider/util/KoFileDeclarationProvider.kt [hunk 1]
```diff
@@ -70,14 +70,22 @@ internal object KoFileDeclarationProvider {
...
                             .toList()
                             .awaitAll()
+                            .filterNotNull()
                     }.also { createKoFilesDeclarationDeferred = it }
                 }
 
             currentDeferred.await()
         }
 
-    private suspend fun parseKotlinFile(it: File): KoFileDeclaration =
```
Comment (koalaparadise) [inline lib/src/main/kotlin/com/lemonappdev/konsist/core/provider/util/KoFileDeclarationProvider.kt:79]: There's no need for this function to be suspending, create a `coroutineScope` and run an `async`. This function is already ran in an `async` in the `map` above.

Comment (chrisbanes) [inline lib/src/main/kotlin/com/lemonappdev/konsist/core/provider/util/KoFileDeclarationProvider.kt:79]: There's no need for this function to be suspending, create a coroutineScope and run an async. This function is already ran in an async in the map above.

**Review by igorwojda — APPROVED** [review]
Thx for the PR and solid explanation 🙏

**Comment by chrisbanes** [comment]
Konsist does ignore them, but AFTER `getKoFileDeclarations` is called (see [here](https://github.com/chrisbanes/konsist/blob/696b67799655e2154447ab45f748e983d8bcc1b5/lib/src/main/kotlin/com/lemonappdev/konsist/core/container/KoScopeCreatorCore.kt#L79)).

I did think about moving that `filter` into the filetree walk, but wasn't sure if there was a reason why it is done after.

**Comment by igorwojda** [comment]
Thx. I will investigate this further

**Pull request merged**